pandas/tseries/tests/test_timeseries.py

```diff
@@ -2454,6 +2455,83 @@ def test_intersection_bug_1708(self):
         result = index_1 & index_2
         self.assertEqual(len(result), 0)
 
+    # GH 10699
+    def test_datetime64_with_DateOffset(self):
+        for klass in (Series, DatetimeIndex):
+            s = klass(date_range('2000-01-01', '2000-01-31'))
+            result = s + pd.DateOffset(years=1)
+            result2 = pd.DateOffset(years=1) + s
+            exp = klass(date_range('2001-01-01', '2001-01-31'))
+            self.assert_numpy_array_equal(result, exp)
```

Comment:
these should all use ``assert_series_equal`` (so construct the required Series), rather than using ``assert_numpy_array_equal``, which is not a strong enough guarantee